apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/AppConfiguration.java

```diff
@@ -338,6 +344,23 @@ static AppConfiguration readConfigFromEnvironmentVariables() {
               "Missing value for configuration environment variable '%s'.",
               ENV_VAR_KEY_DATABASE_PASSWORD));
 
+    String databaseMaxPoolSizeText = System.getenv(ENV_VAR_KEY_DATABASE_MAX_POOL_SIZE);
+    Optional<Integer> databaseMaxPoolSize;
+    if (databaseMaxPoolSizeText == null || databaseMaxPoolSizeText.isEmpty()) {
+      databaseMaxPoolSize = Optional.empty();
+    } else {
+      try {
+        databaseMaxPoolSize = Optional.of(Integer.parseInt(databaseMaxPoolSizeText));
+      } catch (NumberFormatException e) {
+        databaseMaxPoolSize = Optional.of(-1);
+      }
+      if (databaseMaxPoolSize.get() < 1)
+        throw new AppConfigurationException(
+            String.format(
+                "Invalid value for configuration environment variable '%s': '%s'",
+                ENV_VAR_KEY_DATABASE_MAX_POOL_SIZE, databaseMaxPoolSizeText));
+    }
+
```

Comment:
if this new block were moved below the loader threads determination then I think you could avoid the Optional and encapsulate the default case of 'loaderthreads * 2' into this section of code rather than using the .orElse() when building the databaseOptions.

Reply:
True, and very good catch. On reflection, though, I think I prefer the current `Optional` approach as it makes the... intent (?) of the thing a bit clearer to me. Happy to be outvoted, though.

Reply:
I would at least hope for consolidating the logic in one block whether an optional is used or not so that someone can tell by looking at one piece of code how the value is arrived at.

Reply:
Excellent suggestion, and done.